File: presentation/src/main/java/com/github/chojmi/inspirations/presentation/model/gallery/Person.java
package com.github.chojmi.inspirations.presentation.model.gallery;

import android.os.Parcelable;

import com.google.auto.value.AutoValue;

@AutoValue
public abstract class Person implements Parcelable {
    public static Person create(String username, String iconUrl) {
        return new AutoValue_Person(username, iconUrl);
    }

    public abstract String getUsername();

    public abstract String getIconUrl();
}

File: presentation/src/main/java/com/github/chojmi/inspirations/presentation/gallery/model/Person.kt
package com.github.chojmi.inspirations.presentation.gallery.model

import paperparcel.PaperParcel
import paperparcel.PaperParcelable

@PaperParcel
data class Person(val id: String, val username: String, val iconUrl: String) : PaperParcelable {
    companion object {
        @JvmField val CREATOR = PaperParcelPerson.CREATOR
    }
}